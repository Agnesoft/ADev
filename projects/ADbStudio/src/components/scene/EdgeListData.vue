<template>   
    <ListData :data="edges" v-on="$listeners"/>
</template>

<script>
import { mapGetters } from 'vuex'
import ListData from './ListData';

export default {
    name: "EdgeListData",
    components: {
        ListData,
    },
    computed: {
        ...mapGetters({
            edges: 'scene/getEdges',
        }),
    },
}
</script>

<style scoped>

</style>
