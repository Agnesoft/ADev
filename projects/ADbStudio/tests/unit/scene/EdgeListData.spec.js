import { mount  } from '@vue/test-utils';
import Vuex from 'vuex';
import EdgeListData from "@/components/scene/EdgeListData";
import scene_store from '../../../src/store/modules/scene'

import { localVue, i18n,scene_state } from '../TestUtils'


describe('EdgeListData',() => {
    let wrapper;
    let store;

    beforeAll(() => {
        
        localVue.use(Vuex)
        store = new Vuex.Store({
            modules: {
                scene: {
                    namespaced: true,
                    state: scene_state,
                    getters: scene_store.getters
                }
              }
        });
        
        wrapper = mount(EdgeListData, {
            localVue,
            i18n,
            store,
        });
    })
    it('renders all elements', () => {
        expect(wrapper.findAll('li').at(0).text()).toBe("-1");
    })
})
