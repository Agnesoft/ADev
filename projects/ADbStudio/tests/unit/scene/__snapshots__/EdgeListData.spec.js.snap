// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EdgeListData renders correctly (snapshot) 1`] = `
<ul
  class="list-data"
>
  <li>
    <div
      class="element"
    >
      
    -1

    </div>
  </li>
</ul>
`;
